dhctl/pkg/config/meta_test.go

```diff
@@ -300,3 +315,65 @@ func TestParseRegistryData(t *testing.T) {
 		})
 	})
 }
+
+func TestEnrichProxyData(t *testing.T) {
+	t.Run("proxy config is absent", func(t *testing.T) {
+		cfg := generateMetaConfig(t, map[string]interface{}{})
+
+		p, err := cfg.EnrichProxyData()
+		require.NoError(t, err)
+
+		require.Equal(t, p, map[string]interface{}(nil))
+	})
+
+	t.Run("proxy config is present, httpProxy is set", func(t *testing.T) {
+		cfg := generateMetaConfig(t, map[string]interface{}{
+			"proxy": map[string]interface{}{
+				"httpProxy": "http://1.2.3.4",
+			},
+		})
+
+		p, err := cfg.EnrichProxyData()
+		require.NoError(t, err)
+
+		require.Equal(t, p, map[string]interface{}{
+			"httpProxy": "http://1.2.3.4",
+			"noProxy":   []string{"127.0.0.1", "169.254.169.254", "cluster.local", "10.111.0.0/16", "10.222.0.0/16"},
+		})
+	})
+
+	t.Run("proxy config is present, httpsProxy is set", func(t *testing.T) {
+		cfg := generateMetaConfig(t, map[string]interface{}{
+			"proxy": map[string]interface{}{
+				"httpsProxy": "http://1.2.3.4",
+			},
+		})
+
+		p, err := cfg.EnrichProxyData()
+		require.NoError(t, err)
+
+		require.Equal(t, p, map[string]interface{}{
+			"httpsProxy": "http://1.2.3.4",
+			"noProxy":    []string{"127.0.0.1", "169.254.169.254", "cluster.local", "10.111.0.0/16", "10.222.0.0/16"},
+		})
+	})
+
+	t.Run("proxy config is present, all options is set", func(t *testing.T) {
+		cfg := generateMetaConfig(t, map[string]interface{}{
+			"proxy": map[string]interface{}{
+				"httpProxy":  "http://1.2.3.4",
+				"httpsProxy": "http://1.2.3.4",
```

Comment:
please use different values for different fields

Reply:
done